Makefile

```diff
@@ -191,6 +191,7 @@ RELEASE_TAG := $(shell git describe --abbrev=0 2>/dev/null)
 .PHONY: release
 release:  ## Builds and push container images using the latest git tag for the commit.
 	@if [ -z "${RELEASE_TAG}" ]; then echo "RELEASE_TAG is not set"; exit 1; fi
+	git checkout ${RELEASE_TAG}
```

Comment:
xref https://github.com/kubernetes-sigs/cluster-api/pull/1363#issuecomment-527590132... If we're going to do this, we may want to notify the user and/or ask for permission to proceed. We also should make sure the working tree is clean (no modified files, no untracked files).

Reply:
And we should probably change back to whatever HEAD they had before

Reply:
That seems a little overkill for the target, we can add these notes to the documentation, WDYT?

Reply:
I at least want something that checks for a clean checkout

Reply:
PTAL, I added a check to ask user for confirmation